CppCoreGuidelines.md

```diff
@@ -9994,7 +9993,7 @@ Using `unique_ptr` in this way both documents and enforces the function call's r
 * (Simple) Warn if a function takes a `Unique_pointer<T>` parameter by lvalue reference and does not either assign to it or call `reset()` on it on at least one code path. Suggest taking a `T*` or `T&` instead.
 * (Simple) ((Foundation)) Warn if a function takes a `Unique_pointer<T>` parameter by reference to `const`. Suggest taking a `const T*` or `const T&` instead.
 
-### <a name="Rr-sharedptrparam-owner"></a>R.34: Take a `shared_ptr<widget>` parameter to express shared ownership
+### <a name="Rr-sharedptrparam-owner"></a>R.34: Take a `shared_ptr<widget>&&` or `const shared_ptr<widget>&` parameter to express shared ownership
```

Comment:
Continue to allow pass by value?  It is arguable that a shared_ptr is "cheap to copy".  Also rule F.15 says to keep parameter passing simple.

Reply:
Thanks.  Could we get an official ruling on whether `shared_ptr` is "cheap to copy"?  This PR is assuming it was not cheap to copy, since copying incurs an "acquire-release" synchronization in the destructor.  I don't have a strong opinion here but I believe it is ambiguous enough that we should have an official call about this in the guidelines.

Reply:
I'd vote for cheap to copy.  In practice, the performance of passing a shared pointer wont matter because typically the time spent using the contents of the shared pointer will be much larger than the time spent passing it.  Also the ref count change can be avoided if the caller moves it's shared pointer.

Reply:
I have no strong opinion.  I can see your argument but there are functions that are so cheap that the copy overhead could matter.  The current guidance for cheap to copy types tends to imply only types where the overhead is so low as to “never matter” should count.  Would it help matters if I wrote an alternative PR from the “shared_ptr is cheap to copy” point of view?  My main goal here is to have a consistent set of guidelines.